### PR TITLE
Strava App Integration in MindsDB

### PR DESCRIPTION
**Pull request by bala-ceg**
## Description
Implemented the MindsDB handler as per the new application handler framework

**Fixes** #5648

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📄 This change requires a documentation update

## What is the solution?
- [x] Strava all_clubs table 
  - [x] Support LIMIT
  - [x] Support ORDER BY
  - [x] Support column selection
  - [x] Implemented Pagination support - requests 100 records per page

- [x] Strava club_activities table 
  - [x] Support LIMIT
  - [x] Support WHERE
  - [x] Support ORDER BY
  - [x] Support column selection
  - [x] Implemented Pagination support - requests 100 records per page

### Working Screenshots
Screenshot 2023-04-23 at 10 45 48 AM
Screenshot 2023-04-23 at 10 48 31 AM
Screenshot 2023-04-23 at 10 54 44 AM
Screenshot 2023-04-23 at 11 02 01 AM

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation, or created issues to update them.
- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] I have shared a short loom video or screenshots demonstrating any new functionality.

### CODE REVIEW

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import pandas as pd
+import json
+
+logger = get_log("integrations.stravahandler")
+
+class StravaHandler(APIHandler):
+    """Strava handler implementation"""
+
+    def __init__(self, name=None, **kwargs):
+        """Initialize the Confluence handler.
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/strava_handler/strava_handler.py:20]: Please change this

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        ----------
+        name : str
+            name of a handler instance
+        """
+        super().__init__(name)
+
+        connection_data = kwargs.get("connection_data", {})
+
+        self.parser = parse_sql
+        self.dialect = 'strava'
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/strava_handler/strava_handler.py:31]: This is non existing dialect, better to remove it

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

**Comment by ZoranPandovski** [comment]
@bala-ceg Wouldn't it be better if we use https://github.com/stravalib/stravalib instead of direct REST calls?

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            connection object
+        """
+        if self.is_connected is True:
+            return self.connection
+
+        strava_url = f"https://www.strava.com/api/v3/athlete/clubs"
+        headers = {
+            "Authorization": f"Bearer {self.connection_data['strava_api_token']}",
+        }
+        response = requests.request("GET",strava_url,headers=headers)
```
Comment (StpMax) [inline mindsdb/integrations/handlers/strava_handler/strava_handler.py:57]: please add a timeout to any call of 'requests' (5 or 10 seconds?)

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            query in a native format
+        Returns
+        -------
+        StatusResponse
+            Request status
+        """
+        ast = parse_sql(query, dialect="mindsdb")
+        return self.query(ast)
+
+
```
Comment (StpMax) [inline mindsdb/integrations/handlers/strava_handler/strava_handler.py:102]: please add also `get_tables` and `get_columns` methods,  `connection_args` and `connection_args_example` dicts

**Review by StpMax — CHANGES_REQUESTED** [review]

**Comment by bala-ceg** [comment]
@ZoranPandovski @StpMax please advise whether I should change this implementation using stravalib python library or should I just continue this with rest API calls

**Comment by ZoranPandovski** [comment]
I think it would be easily maintainable and we can include support for more endpoints if we use the library. Please check  it

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py (hunk 4 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/strava_handler/strava_handler.py]: @StpMax added the connection_args & connection_args_example dicts. I have implemented get_columns in strava_tables.py & get_tables method is replaced by  call_strava_allclubs_api() or call_strava_clubactivities_api() as per the application handler framework
Please let me know if i am missing something. Thank you

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py (hunk 3 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/strava_handler/strava_handler.py]: Dropped requests & implemented the handler with strava python API lib

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py (hunk 2 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/strava_handler/strava_handler.py]: Removed it. thanks zoran for mentioning this

File: mindsdb/integrations/handlers/strava_handler/strava_handler.py (hunk 1 above)
Comment (bala-ceg) [reply mindsdb/integrations/handlers/strava_handler/strava_handler.py]: Changed this. Thanks

**Comment by bala-ceg** [comment]
@ZoranPandovski @StpMax Please review the handler, i have addressed the review comments in my new changes

**Review by StpMax — APPROVED** [review]

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**